src/EmailServiceClient.php

```diff
@@ -0,0 +1,260 @@
+            );
+        }
+
+        $this->initializeConfig($config);
+
+        // Create the client (applying some defaults).
+        parent::__construct(array_replace_recursive([
+            'description_path' => \realpath(
+                __DIR__ . '/descriptions/email-service-api.php'
+            ),
+            'description_override' => [
+                'baseUri' => $baseUri,
+            ],
+            'access_token' => $accessToken,
+            'http_client_options' => [
+                'timeout' => 30,
+            ],
+        ], $config));
+    }
+
+    /*
+     * Validates the config values for ASSERT_VALID_IP_CONFIG and
+     *   ASSERT_VALID_IP_CONFIG
+     * Uses them to set $this->assertValidIp and $this->trustedIpRanges
+     *
+     * @param array the config values for the client
+     *
+     * @return null
+     * @throws \InvalidArgumentException
+     */
+    private function initializeConfig($config)
+    {
+
+        if (isset($config[self::ASSERT_VALID_IP_CONFIG])) {
+            $this->assertValidIp = $config[self::ASSERT_VALID_IP_CONFIG];
+        }
+
+        // If we don't need to validate the service Ip, we're done here
+        if ( ! $this->assertValidIp) {
+            return;
+        }
+
+        /*
+         *  If we should validate the service IP but there aren't
+         *  any trusted IPs, throw an exception
+         */
+        if (empty($config[self::TRUSTED_IPS_CONFIG])) {
+            throw new \InvalidArgumentException(
+                'The config entry for ' . self::TRUSTED_IPS_CONFIG .
+                ' must be set (as an array) when ' .
+                self::ASSERT_VALID_IP_CONFIG .
+                ' is not set or is set to True.',
+                1494531150
+            );
+        }
+
+        /*
+         * At this point, we need to validate the service Ip and we know
+         * that the TRUSTED_IPS_CONFIG is not empty
+         */
+        $newTrustedIpRanges = $config[self::TRUSTED_IPS_CONFIG];
+        if ( ! is_array($newTrustedIpRanges)) {
+            throw new \InvalidArgumentException(
+                'The config entry for ' . self::TRUSTED_IPS_CONFIG .
+                ' must be an array.',
+                1494531200
+            );
+        }
+
+        foreach ($newTrustedIpRanges as $nextIpRange) {
+            $ipBlock = IPBlock::create($nextIpRange);
+            $this->trustedIpRanges[] = $ipBlock;
+        }
+
+        $this->assertTrustedIp();
+    }
+
+    /**
+     * Attempt to authenticate using the given credentials, getting back
+     * information about the authenticated user (if the credentials were
+     * acceptable) or null (if unacceptable).
+     *
+     * @param string $username The username.
+     * @param string $password The password (in plaintext).
+     * @return array|null An array of user information (if valid), or null.
+     * @throws Exception
+     */
+    public function authenticate(string $username, string $password)
```

Comment:
I think this whole function can be deleted, as it seems to be a carryover from the ID Broker PHP Client that isn't applicable here.